jwst/assign_mtwcs/moving_target_wcs.py

```diff
@@ -148,3 +141,39 @@ def add_mt_frame(wcs, ra_average, dec_average, mt_ra, mt_dec):
     pipeline.append((mt, None))
     new_wcs = WCS(pipeline)
     return new_wcs
+
+
+def _is_mt_meta_valid(meta):
+    """
+    Check if the metadata contains valid moving target RA/DEC.
+
+    Includes a check for all slits in a MultiSlitModel. The metadata should contain
+    slit information in keys of the form 'slits.<n>.meta.wcsinfo.mt_ra' and
+    'slits.<n>.meta.wcsinfo.mt_dec', where <n> is the slit index.
+
+    Parameters
+    ----------
+    meta : dict
+        Flattened metadata dictionary from a data model, as output from `read_metadata`.
+
+    Returns
+    -------
+    bool
+        True if valid, False otherwise.
+    """
+    # find each slit's meta.wcsinfo for MultiSlitModel types.
+    # For other input types, n_slits is 0 and mt_ra_keys and mt_dec_keys
+    # are expected to have length 1
+    slit_meta = [key for key in meta.keys() if key.startswith("slits.")]
```

Comment:
Is there an issue (or issues) addressed by this new code? It looks quite different than the replaced code.